go/database/mpt/io/parallel_visit_test.go

```diff
@@ -219,45 +219,68 @@ func TestVisit_Nodes_CannotOpenFiles(t *testing.T) {
 }
 
 func TestVisit_Nodes_CannotCloseSources(t *testing.T) {
-	injectedError := fmt.Errorf("injected error")
-
-	trie := createArchive(t)
-
-	nodeId, err := trie.GetBlockRoot(0)
-	if err != nil {
-		t.Fatalf("failed to get block root: %v", err)
-	}
-	if err := trie.Close(); err != nil {
-		t.Fatalf("failed to close archive: %v", err)
+	sourceFactories := []struct {
+		name       string
+		factory    func(dir string) nodeSourceFactory
+		createData func(t *testing.T, dir string) (root mpt.NodeId)
+	}{
+		{"hashWithNodes", func(dir string) nodeSourceFactory { return &nodeSourceHashWithNodesFactory{directory: dir} }, func(t *testing.T, dir string) (root mpt.NodeId) {
+			trie := createArchive(t, dir)
+			root, err := trie.GetBlockRoot(0)
+			if err != nil {
+				t.Fatalf("failed to get block root: %v", err)
+			}
+			if err := trie.Close(); err != nil {
+				t.Fatalf("failed to close archive: %v", err)
+			}
+			return root
+		}},
```

Comment:
nit: please try to keep the line-width low (e.g. below 120 characters) to facilitate reviews also on laptops if needed. Here it should be doable by doing a line break for each parameter of the struct and the function bodies.

```suggestion
		{
			"hashWithNodes",
			func(dir string) nodeSourceFactory {
				return &nodeSourceHashWithNodesFactory{directory: dir}
			},
			func(t *testing.T, dir string) (root mpt.NodeId) {
				trie := createArchive(t, dir)
				root, err := trie.GetBlockRoot(0)
				if err != nil {
					t.Fatalf("failed to get block root: %v", err)
				}
				if err := trie.Close(); err != nil {
					t.Fatalf("failed to close archive: %v", err)
				}
				return root
			},
		},
```

Reply:
sorry, it must have been some auto-formating which I have missed.